wgsl/index.bs

```diff
@@ -6970,17 +6965,11 @@ We note "*s*: *B*" to say that *s* respects the rules regarding behaviors, and h
 
 For each function:
 - Its body [=shader-creation error|must=] be a valid statement by these rules.
-- If the function has a return type, the [=behavior=] of its body [=shader-creation error|must=] be one of {Return} or {Return, Discard}.
-- Otherwise, the [=behavior=] of its body [=shader-creation error|must=] be a subset of {Next, Return, Discard}.
+- If the function has a return type, the [=behavior=] of its body [=shader-creation error|must=] be {Return}.
+- Otherwise, the [=behavior=] of its body [=shader-creation error|must=] be a subset of {Next, Return}.
 
 We assign a [=behavior=] to each function: it is its body's [=behavior=] (treating the body as a regular statement), with any "Return" replaced by "Next".
-As a consequence of the rules above, a function behavior is always one of {}, {Next}, {Discard}, or {Next, Discard}.
-
-Similarly, we assign a [=behavior=] to each expression, since expressions can include function calls, which can discard.
-Like functions, expression behaviors are always one of {}, {Next}, {Discard}, or {Next, Discard}.
-
-Note: There is currently no valid program with an expression that does not have Next in its [=behavior=].
-The reason is that only functions without a return type can have such a [=behavior=], and there is no compound expression in which such a function can be called.
+As a consequence of the rules above, a function behavior is always one of {}, or {Next}.
```

Comment:
I got sidetracked wondering why the analysis of expressions went away.
Is it worth a note here saying the {} behaviour is invalid?  (I know the rule is below, down at the end of the section.)

Reply:
I shuffled the later rules a bit and moved the requirement about non-empty behaviour eariler.